lib/src/Uncertainty/Distribution/openturns/BlockIndependentDistribution.hxx

```diff
@@ -0,0 +1,183 @@
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_BLOCKINDEPENDENTDISTRIBUTION_HXX
+#define OPENTURNS_BLOCKINDEPENDENTDISTRIBUTION_HXX
+
+#include "openturns/DistributionImplementation.hxx"
+#include "openturns/PersistentCollection.hxx"
+#include "openturns/Distribution.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class ComposedCopula
+ *
+ * The class describes the probabilistic concept of copulas
+ * made from a collection of copulas joined by an independent copula
+ */
+class OT_API BlockIndependentDistribution
+  : public DistributionImplementation
+{
+  CLASSNAME
+public:
+
+  /** A type for distribution collection */
+  typedef Collection<Distribution>                 DistributionCollection;
+  typedef PersistentCollection<Distribution>       DistributionPersistentCollection;
+
+  /** Default constructor for save/load methods : 1D distribution with default Uniform marginal and IndependentCopula */
+  BlockIndependentDistribution();
+
+  /** Default constructor */
+  explicit BlockIndependentDistribution(const DistributionCollection & coll);
+
+  /** Comparison operator */
+  Bool operator ==(const BlockIndependentDistribution & other) const;
+protected:
+  Bool equals(const DistributionImplementation & other) const;
+public:
+
+  /** String converter */
+  String __repr__() const;
+  String __str__(const String & offset = "") const;
+
+
+  /** Copula collection accessor */
+  void setDistributionCollection(const DistributionCollection & coll);
+  const DistributionCollection & getDistributionCollection() const;
+
+  /* Here is the interface that all derived class must implement */
+
+  /** Virtual constructor */
+  virtual BlockIndependentDistribution * clone() const;
+
+  /** Get one realization of the BlockIndependentDistribution */
+  Point getRealization() const;
+
+  /** Get the DDF of the BlockIndependentDistribution */
+  using DistributionImplementation::computeDDF;
+  Point computeDDF(const Point & point) const;
+
+  /** Get the PDF of the BlockIndependentDistribution */
+  using DistributionImplementation::computePDF;
+  Scalar computePDF(const Point & point) const;
+
+  /** Get the log-PDF of the ComposedCopula */
+  using DistributionImplementation::computeLogPDF;
+  Scalar computeLogPDF(const Point & point) const;
+
+  /** Get the CDF of the ComposedCopula */
+  using DistributionImplementation::computeCDF;
+  Scalar computeCDF(const Point & point) const;
+
+  /** Get the probability content of an interval */
+  Scalar computeProbability(const Interval & interval) const;
+
+  /** Compute the survival function */
+  using DistributionImplementation::computeSurvivalFunction;
+  Scalar computeSurvivalFunction(const Point & point) const;
+
+  /** Get the Spearman correlation of the distribution */
+  CorrelationMatrix getSpearmanCorrelation() const;
+
```

Comment:
We have here `getSpearmanCorrelation`and `getKendallTau`but no `getCorrelation`. An omission?

Reply:
No, it is because the generic implementation of getCorrelation() relies on computeCovariance(), which has already been overloaded in the class. The gain in terms of speed would not be significant.

Reply:
ok